library/src/cache/patch_manager.rs

```diff
@@ -817,10 +889,13 @@ mod next_boot_patch_tests {
         let temp_dir = TempDir::new("patch_manager")?;
         let mut manager = PatchManager::manager_for_test(&temp_dir);
 
-        // Add a first patch and pretend it booted successfully.
+        // Add a first patch and record that we started booting it, but not that it succeeded.
         manager.add_patch_for_test(&temp_dir, 1)?;
         manager.record_boot_start_for_patch(1)?;
 
+        // Simulate a restart.
```

Comment:
This is really simulating a second start of the app.  e.g. the app crashed and is being started again.

Reply:
Updated test (and other similar tests) to be clearer